external-crates/move/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -128,7 +127,7 @@ impl PartialEq for Package {
         // comparing packages during insertion of externally resolved ones - an internally resolved
         // existing package in the graph would not be recognized as a potential different version of
         // the externally resolved one)
-        self.kind == other.kind && self.version == other.version
```

Comment:
High-level comment on this commit: I'm a big fan of putting up this PR separately before everything else since it's clean / self-contained. We need to cover our bases with how this affects users / manifests that contain `version`. I.e., great if it's vestigial and we can conceptually remove it. To go along with that change, for example, we need to remove references to `version` in the sui repo in `examples` and elsewhere: https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/+file:Move%5C.toml+version+-repo:diem+-repo:aptos&patternType=standard&sm=1&groupBy=repo and communicate the change in the changelog

Reply:
> To go along with that change, for example, we need to remove references to version in the sui repo in examples and elsewhere

As far as I'm concerned we can do that but we need to consider is that old versions of the tool will then error when building packages against master because version field is mandatory.